File: Strings/Excel_Sheet_Column_Number/ExcelSheetColumnNumber.py
class Solution:
    def titleToNumber(self, columnTitle: str) -> int:
        result = 0
        for char in columnTitle:
            result = result * 26 + (ord(char.upper()) - ord('A') + 1)
        return result

# Taking user input
if __name__ == "__main__":
    columnTitle = input("Enter Excel column title (e.g., 'AB'): ")
    solution = Solution()
    print("Column Number is:", solution.titleToNumber(columnTitle))

File: Strings/Excel_Sheet_Column_Number/ExcelSheetColumnNumber.java
import java.util.Scanner;

public class Solution {
    public int titleToNumber(String columnTitle) {
        int result = 0;
        for (int i = 0; i < columnTitle.length(); i++) {
            result = result * 26 + (columnTitle.charAt(i) - 'A' + 1);
        }
        return result;
    }

    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);
        System.out.print("Enter Excel column title (e.g., 'AB'): ");
        String columnTitle = sc.nextLine();
        Solution solution = new Solution();
        System.out.println("Column Number is: " + solution.titleToNumber(columnTitle));
        sc.close();
    }
}
